hops/featurestore_impl/dao/common/featurestore_metadata.py

```diff
@@ -50,14 +50,14 @@ def _parse_featurestore_metadata(self, metadata_json):
         features_to_featuregroups = {}
         storage_connectors = {}
         for fg in metadata_json[constants.REST_CONFIG.JSON_FEATUREGROUPS]:
+            fg_obj = Featuregroup(fg)
             featuregroups[fs_utils._get_table_name(fg[constants.REST_CONFIG.JSON_FEATUREGROUP_NAME],
-                                                     fg[constants.REST_CONFIG.JSON_FEATUREGROUP_VERSION])] = \
-                Featuregroup(fg)
+                                                     fg[constants.REST_CONFIG.JSON_FEATUREGROUP_VERSION])] = fg_obj
             for f in fg[constants.REST_CONFIG.JSON_FEATUREGROUP_FEATURES]:
                 if f[constants.REST_CONFIG.JSON_FEATURE_NAME] in features_to_featuregroups:
-                    features_to_featuregroups[f[constants.REST_CONFIG.JSON_FEATURE_NAME]].append(Featuregroup(fg))
+                    features_to_featuregroups[f[constants.REST_CONFIG.JSON_FEATURE_NAME]].append(Featuregroup(fg_obj))
```

Comment:
I think you were a bit too quick here, shouldn't it be:
```suggestion
                    features_to_featuregroups[f[constants.REST_CONFIG.JSON_FEATURE_NAME]].append((g_obj)
```